src/steps/account.ts

```diff
@@ -16,33 +16,36 @@ export async function fetchAccountDetails({
 
   const data = await apiClient.getAccountDetails();
 
-  await jobState.addEntity(
-    createIntegrationEntity({
-      entityData: {
-        source: data,
-        assign: {
-          _type: 'fastly_account',
-          _class: 'Account',
-          _key: `fastly-account:${data.id}`,
-          id: data.id,
-          displayName: data.name || data.id,
-          name: data.name,
-          address: data.postal_address,
-          phone: data.phone_number,
-          mfaEnabled: data.force_2fa,
-          mfaEnforced: data.force_2fa,
-          ssoEnforced: data.force_sso,
-          readOnly: data.readonly,
-          ownerId: data.owner_id,
-          ipWhitelist: data.ip_whitelist,
-          rateLimit: data.rate_limit,
-          createdOn: getTime(data.created_at),
-          updatedOn: getTime(data.updated_at),
-          deletedOn: getTime(data.deleted_at),
-        },
+  const accountEntity = createIntegrationEntity({
+    entityData: {
+      source: data,
+      assign: {
+        _type: 'fastly_account',
+        _class: 'Account',
+        _key: `fastly-account:${data.id}`,
+        id: data.id,
+        displayName: data.name || data.id,
+        name: data.name,
+        address: data.postal_address,
+        phone: data.phone_number,
+        mfaEnabled: data.force_2fa,
+        mfaEnforced: data.force_2fa,
+        ssoEnforced: data.force_sso,
+        readOnly: data.readonly,
+        ownerId: data.owner_id,
+        ipWhitelist: data.ip_whitelist,
+        rateLimit: data.rate_limit,
+        createdOn: getTime(data.created_at),
+        updatedOn: getTime(data.updated_at),
+        deletedOn: getTime(data.deleted_at),
       },
-    }),
-  );
+    },
+  });
+
+  await Promise.all([
+    jobState.addEntity(accountEntity),
+    jobState.setData(accountEntity._key, accountEntity),
```

Comment:
To avoid having to know the `_key` value later, consider doing something like `jobState.setData(ACCOUNT_ENTITY_DATA_KEY, accountEntity)`, which allows later `jobState.getData(ACCOUNT_ENTITY_DATA_KEY)`.

That said, if you know the `_key`, you can avoid this `set/getData` thing and later just do `jobState.findEntity(known_key)`.

Just to clarify, the goal is to avoid knowing/constructing `_key` values in code in multiple places unless absolutely necessary, to make refactoring easier/less prone to errors. We have plans to make the `findEntity` allow for using something other than just `_key` in the future, something along the lines of what [we plan for `iterateEntities`](https://github.com/JupiterOne/sdk/issues/79). In this case, you should be able to do `findEntity({ _class: 'Account' })`.